mitmproxy-rs/src/dns_resolver.rs

```diff
@@ -0,0 +1,19 @@
+use hickory_resolver::config::{LookupIpStrategy, ResolverConfig, ResolverOpts};
+use hickory_resolver::TokioAsyncResolver;
+use pyo3::prelude::{pyfunction, PyResult, Python};
+use pyo3::types::PyAny;
+
+#[pyfunction]
+#[pyo3(signature = (host, is_ipv6))]
+pub fn getaddrinfo(py: Python<'_>, host: String, is_ipv6: bool) -> PyResult<&PyAny> {
+    pyo3_asyncio::tokio::future_into_py(py, async move {
+        let mut opts = ResolverOpts::default();
+        if is_ipv6 {
+            opts.ip_strategy = LookupIpStrategy::Ipv6Only;
+        }
```

Comment:
Is there a difference between using this method versus quering for `AAAA` record using the `lookup` method directly?

Reply:
Not sure if I undserstand the question?

Reply:
We can either use the current method or use `resolver.lookup(domain, RecordType::AAAA)` to query `AAAA` records, but now after looking at https://github.com/mitmproxy/mitmproxy_rs/pull/148#discussion_r1660180768, I think we should prefer the current method